### PR TITLE
make imports work in python3

### PR DESCRIPTION
**Pull request by pbauer**

### CODE REVIEW

**Comment by pbauer** [comment]
A valid question. There is no clear recommendation about how to deal with python3 suppport  at the moment. Many zope packages have a `_compat.py` but we depend on six anyway, so we could say for example that those packages that depend on CMFPlone should use six and others may do as seems prudent. But maybe there are other opinions. 

We will document the recommendations in https://docs.google.com/document/d/1T-E6qD7BoTrebk9L6F1EC0qMvqSOAu0987av6wUZ7nY/edit#

**Comment by tomgross** [comment]
Zope itself depens on six https://github.com/zopefoundation/Zope/blob/master/setup.py#L73 so I guess it can safely be assumed it is there everywhere Plone. I also would preferre six in packages not depending on Zope for sake of unity (think of plone.api). I don't see any advantages in NOT using `six`.

**Comment by gforcada** [comment]
@pbauer is this still needed or @ale-rt pull requests already cover everything?

**Comment by ale-rt** [comment]
Some patches are still relevant!
This will be a long journey ;)

File: Products/CMFPlone/__init__.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def initialize(context):
     this_module.Batch = Batch
 
     ModuleSecurityInfo('StringIO').declarePublic('StringIO')
-    from StringIO import StringIO
+    from six import StringIO
```
Comment (hvelarde) [inline Products/CMFPlone/__init__.py]: use `from io import StringIO` instead.

File: Products/CMFPlone/controlpanel/tests/test_controlpanel_browser_usergroups_siteadmin_role.py [hunk 2]
```diff
@@ -4,18 +4,17 @@
...
    CMFPlone fixture at some point.
 """
-from cStringIO import StringIO
-from urllib import urlencode
+from plone.app import testing
+from plone.app.testing.bbb import PloneTestCase as FunctionalTestCase
+from plone.app.testing.bbb import PloneTestCaseFixture
 from plone.protect.authenticator import createToken
+from Products.CMFCore.utils import getToolByName
+from six import StringIO
```
Comment (hvelarde) [inline Products/CMFPlone/controlpanel/tests/test_controlpanel_browser_usergroups_siteadmin_role.py]: use `from io import StringIO` instead.

File: Products/CMFPlone/resources/browser/combine.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 from plone.resource.interfaces import IResourceDirectory
 from Products.CMFPlone.interfaces import IBundleRegistry
 from Products.CMFPlone.interfaces.resources import OVERRIDE_RESOURCE_DIRECTORY_NAME  # noqa
-from StringIO import StringIO
+from six import StringIO
```
Comment (hvelarde) [inline Products/CMFPlone/resources/browser/combine.py]: use `from io import StringIO` instead.

File: Products/CMFPlone/resources/browser/cook.py [hunk 4]
```diff
@@ -9,8 +9,8 @@
 from Products.CMFPlone.interfaces.resources import OVERRIDE_RESOURCE_DIRECTORY_NAME  # noqa
 from Products.CMFPlone.resources.browser.combine import combine_bundles
 from scss import Compiler
+from six import StringIO
```
Comment (hvelarde) [inline Products/CMFPlone/resources/browser/cook.py]: use `from io import StringIO` instead.

File: Products/CMFPlone/tests/dummy.py [hunk 5]
```diff
@@ -2,23 +2,18 @@
...
-from Products.CMFPlone.interfaces import INonStructuralFolder
-from Products.CMFPlone.interfaces import IWorkflowChain
-
 from ComputedAttribute import ComputedAttribute
-from OFS.SimpleItem import SimpleItem
 from OFS.Folder import Folder as SimpleFolder
+from OFS.SimpleItem import SimpleItem
+from Products.CMFPlone.interfaces import INonStructuralFolder
+from Products.CMFPlone.interfaces import IWorkflowChain
+from six import StringIO
```
Comment (hvelarde) [inline Products/CMFPlone/tests/dummy.py]: use `from io import StringIO` instead.

File: Products/CMFPlone/tests/testCSRFProtection.py [hunk 6]
```diff
@@ -5,7 +5,7 @@
 from plone.app.testing.bbb import PloneTestCase
 from plone.keyring.interfaces import IKeyManager
 from plone.protect.authenticator import AuthenticatorView
-from StringIO import StringIO
+from six import StringIO
```
Comment (hvelarde) [inline Products/CMFPlone/tests/testCSRFProtection.py]: use `from io import StringIO` instead.

File: Products/CMFPlone/tests/testControlPanelScripts.py [hunk 7]
```diff
@@ -1,13 +1,13 @@
...
+from DateTime import DateTime
+from plone.app.testing import SITE_OWNER_NAME
+from plone.app.testing import SITE_OWNER_PASSWORD
 from plone.app.testing import TEST_USER_ID
 from plone.app.testing import TEST_USER_NAME
 from plone.app.testing import TEST_USER_PASSWORD
-from plone.app.testing import SITE_OWNER_NAME
-from plone.app.testing import SITE_OWNER_PASSWORD
 from Products.CMFPlone.tests.PloneTestCase import PloneTestCase
+from six import StringIO
```
Comment (hvelarde) [inline Products/CMFPlone/tests/testControlPanelScripts.py]: use `from io import StringIO` instead.

File: Products/CMFPlone/tests/testWebDAV.py [hunk 8]
```diff
@@ -6,7 +6,7 @@
 from Products.CMFCore.utils import getToolByName
 from Products.CMFPlone.tests import dummy
 from Products.CMFPlone.tests import PloneTestCase
-from StringIO import StringIO
+from six import StringIO
```
Comment (hvelarde) [inline Products/CMFPlone/tests/testWebDAV.py]: use `from io import StringIO` instead.

**Comment by hvelarde** [comment]
in fact, it could be also use `from io import BytesIO`, we need further analysis.

**Comment by ale-rt** [comment]
Using the `io` module is the way to go.
I will also to investigate the same issue to get this one merged https://github.com/zopefoundation/ZServer/pull/7.

**Comment by hvelarde** [comment]
we need a linter or this is going to be a nightmare; any hints?

**Comment by pbauer** [comment]
`from io import StringIO` is obviously wrong: http://jenkins.plone.org/job/pull-request-5.2/82/consoleText

**Comment by jensens** [comment]
Please also delete the branch after merging.

**Pull request merged**